platform/pulp/platform/models/repository.py

```diff
@@ -1,10 +1,268 @@
+from django.db import models
+from django.contrib.contenttypes import fields
 
+from pulp.platform.models import Model, Notes, Scratchpad, Config
 
-class Distributor(Model):
+
+class Content(Model):
     # placeholder
-    pass
+    type = models.TextField()
+    created = models.DateTimeField(auto_now_add=True)
+
+
+class Repository(Model):
+    """
+    Collection of content.
+
+    Fields:
+
+    :cvar name: The consumer name.
+    :type name: models.TextField
+
+    :cvar: description: An optional description.
+    :type: models.TextField
+
+    :cvar type: The repository type.
+    :type type: models.TextField
+
+    :cvar last_content_added: When content was last added.
+    :type last_content_added: models.DateTimeField
+
+    :cvar last_content_removed: When content was last removed.
+    :type last_content_removed: models.DateTimeField
+
+    Relations:
+
+    :cvar scratchpad: Arbitrary information stashed on the repository.
+    :type scratchpad: fields.GenericRelation
+
+    :cvar notes: Arbitrary repository properties.
+    :type notes: fields.GenericRelation
+
+    :cvar content: Associated content.
+    :type content: models.ManyToManyField
+    """
+    name = models.TextField(db_index=True, unique=True)
+    description = models.TextField(blank=True, default='')
+    type = models.TextField()
+
+    last_content_added = models.DateTimeField(blank=True, null=True)
+    last_content_removed = models.DateTimeField(blank=True, null=True)
+
+    scratchpad = fields.GenericRelation(Scratchpad)
+    notes = fields.GenericRelation(Notes)
+
+    content = models.ManyToManyField('Content', through='RepositoryContent')
+
+    @property
+    def content_summary(self):
+        """
+        The contained content summary.
+
+        :return: A dict of {<type>: <count>}
+        :rtype:  dict
+        """
+        mapping = self.content.values('type').annotate(count=models.Count('type'))
+        return {m['type']: m['count'] for m in mapping}
+
+
+class RepositoryGroup(Model):
+    """
+    A group of repositories.
+
+    Fields:
+
+    :cvar name: The group name.
+    :type name: models.TextField
+
+    :cvar: description: An optional description.
+    :type: models.TextField
+
+    Relations:
+    :cvar notes: Arbitrary group properties.
+    :type notes: fields.GenericRelation
+
+    :cvar members: Repositories associated with the group.
+    :type members: models.ManyToManyField
+    """
+    name = models.TextField(db_index=True, unique=True)
+    description = models.TextField(blank=True, default='')
+
+    members = models.ManyToManyField('Repository')
+    scratchpad = fields.GenericRelation(Scratchpad)
+    notes = fields.GenericRelation(Notes)
+
+
+class Plugin(Model):
+    """
+    Abstract model for plugins.
+
+    Fields:
+
+    :cvar type: The plugin name.
+    :type type: models.TextField
+
+    :cvar type: The plugin type.
+    :type type: models.TextField
+
+    :cvar last_updated: When the plugin was last updated.
```

Comment:
@jortel pointed out that there aren't (m?)any mutable fields that aren't config, so I agree there's no need to rename ``last_updated``